### ISSUE TITLE
Sharing Validation where model type is not the same

### ISSUE THREAD

**Issue by emzyme20** [FluentValidation/FluentValidation#1317]
### System Details

- FluentValidation version: 8.1.2
- ASP.NET Core 2.0

### Issue Description

We have AbstractValidator implementations for several types that come into our API and we have an instance where a new type coming into the system wants to share validation from 2 other types as this new type is a combination of the two.. For instance

Take the types:

```csharp
public class PublishCommand
{
}

public class UpdateCommand
{
}

public class UpdateAndPublishCommand
{
}
```

We have the following validators
```csharp
public class PublishValidator : AbstractValidator<PublishCommand>
{
}

public class UpdateValidator : AbstractValidator<UpdateCommand>
{
}
```

What I now need to do is create an `UpdateAndPublishValidator` which would really just like to call into the 2 validators that I have already defined. From my reading and playing around, I don't think this is currently possible using the FluentValidation interface because the `RuleFor` will only run on the type that the validator is defined to use.

### Workaround
The way I am doing it at the moment is to create a static method on the 2 validators that I can call from a new validator:

```csharp
public class UpdateAndPublishValidator : AbstractValidator<UpdateAndPublishCommand>
 {
    public UpdateAndPublishValidator(IProvider provider, IQuery query)
    {
        this.RuleFor(x => x)
            .Custom(
                (cmd, context) =>
                {
                    UpdateValidator.Validate(provider, query, cmd.AggregateId, cmd.UserId, cmd.Type, cmd.PropertiesToUpdate, context);
                    PublishValidator.Validate(queryBus, cmd.ReviewerId, cmd.UserId, context);
                });
    }
 }
```
Is this the recommended way to get this to work or is there a way that I could Register the UpdateValidator and PublishValidator to be able to be run against UpdateAndPublishCommand?

**Comment by JeremySkinner** [comment]
Hi, the answer is "it depends" based on whether you're willing to change the structure of your commands slightly so that they share an inheritance hierarchy. For example, if you introduced some interfaces that contain the shared properties then you could do this:

```csharp
public interface IPublishCommand 
{
   // Properties shared between PublishCommand and UpdateAndPublishCommand in here
}
public interface IUpdateCommand 
{
   // Properties shared between UpdateCommand and UpdateAndPublishCommand in here
}

public class PublishCommand : IPublishCommand
{
}

public class UpdateCommand : IUpdateCommand
{
}

public class UpdateAndPublishCommand : IPublishCommand, IUpdateCommand
{
}
```

If your commands shared an interface in this way, you could then [use the Include feature](https://docs.fluentvalidation.net/en/latest/including-rules.html) to include both the UpdateValidator and PublishValidator within the UpdateAndPublishValidator, eg: 

```csharp
// Note these 2 validators are now validating the interface type. 
public class PublishValidator : AbstractValidator<IPublishCommand> {

}

public class UpdateValidator : AbstractValidator<IUpdateCommand> {

}

// The UpdateAndPublishValidator can now include both of the others, as it shares the same interfaces. 
public class UpdateAndPublishValidator : AbstractValidator<UpdateAndPublishCommand> {
	public UpdateAndPublishValidator() {
		Include(new UpdateValidator());
		Include(new PublishValidator());
	}
}
```

However, if you don't want to introduce shared interfaces in this way then it won't be possible to do what you're asking, which is intentional. FluentValidation is a tool for building strongly-typed rules in a type-safe way and it doesn't allow you to do things that bypass the type safety. If the types are separate and don't share an inheritance hierarcy, then its actually better to explicitly have separate validator classes which duplicate the rules. 

Think of validators as mirroring the types they validate - if the types share an inheritance hierarchy then so can the validators. If they don't share an inheritance hierarchy and just duplicate the property names, then the validators should duplicate their rules too. 

So in summary:
- If you want to introduce interfaces so the types have a shared hierarcy, then yes this is possible using the `Include` mechanism
- If you don't want to do this, then it's actually better not to try and work around it, and instead explicitly have 

Hope that helps.

**Comment by emzyme20** [comment]
Your suggestion to work off of interfaces worked for me. Thank you.
